haystack/query.py

```diff
@@ -233,7 +234,7 @@ def _fill_cache(self, start, end, **kwargs):
         # an array of 100,000 ``None``s consumed less than .5 Mb, which ought
```

Comment:
I would change this comment here explaining that a sparse list is used because the total results set can be many orders of magnitude larger than the set of results that we pull at any given time.